SendGrid/SendGrid/Client.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Text;
+using SendGrid.Resources;
+using System.Web.Script.Serialization;
+using System.Net;
+
+namespace SendGrid
+{
+    public class Client
+    {
+        private string _apiKey;
+        public APIKeys ApiKeys;
+        private Uri _baseUri;
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="apiKey">Your SendGrid API Key</param>
+        /// <param name="baseUri">Base SendGrid API Uri</param>
+        public Client(string apiKey, string baseUri = "https://api.sendgrid.com/")
```

Comment:
Question: why do you provide the option to provide a `baseUri` argument to a different url?  Should that be encapsulated as an implementation detail such that the end-user would never try changing it?